.script/workbooksValidations/previewImageValidator.ts

```diff
@@ -0,0 +1,33 @@
+import { WorkbookValidationError } from "../utils/ValidationError";
+import { WorkbookMetadata } from "../utils/WorkbookMetadata";
+
+const previewImageRegex: RegExp = /(black|white|Black|White)(\d*?).png/;
+const blackPreviewImageRegex: RegExp = /(black|Black)(\d*?).png/;
+const whitePreviewImageRegex: RegExp = /(white|White)(\d*?).png/;
+
+function isAllValidFileNames(previewImagesFileNames: Array<string>): boolean {
+    return previewImagesFileNames.every((previewImageFileName: string) => previewImageRegex.test(previewImageFileName));
+}
+
+function isMissingImages(previewImagesFileNames: Array<string>): boolean {
+    let blackImageCount: number = 0;
+    let whiteImageCount: number = 0;
+    previewImagesFileNames.forEach((filename) => {
+        if (whitePreviewImageRegex.test(filename)) whiteImageCount++;
+        if (blackPreviewImageRegex.test(filename)) blackImageCount++;
+    });
+    return blackImageCount === 0 || whiteImageCount === 0;
+}
+
+export function isValidPreviewImageFileNames(items: Array<WorkbookMetadata>) {
+  items.forEach((workbookMetadata: WorkbookMetadata) => {
+    if (!isAllValidFileNames(workbookMetadata.previewImagesFileNames)) {
+      throw new WorkbookValidationError(`Invalid Preview Images for workbook ${workbookMetadata.workbookKey}. Filename must contain either "Black" or "White" and must end with .png`);
```

Comment:
Technically you can have have Black or White and .png like this errror says and still not match the regexes like `AwsBlackExample.png`
It could be worth putting a more specific message so a 3P knows specifically what to fix.
I guess the format also can include numbers in the filename too like `AwsBlack1.png`


Another option is maybe just split `isAllValidFileNames` into a fewer more simple tests:
- file extension is .png
- filename ends with a number, or nothing
- filename ends with black/white ignoring the number, or just contains black/white.  Not sure the exact requirement

Reply:
I realized the requirement is that file names need to *include* either white or black, so I fixed this. also split the png validation to a separate function as you suggested. Thanks for the catch!